homeassistant/components/goodwe/sensor.py

```diff
@@ -23,19 +24,24 @@
     POWER_WATT,
     TEMP_CELSIUS,
 )
-from homeassistant.core import HomeAssistant
+from homeassistant.core import HomeAssistant, callback
 from homeassistant.helpers.entity import DeviceInfo, EntityCategory
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.event import async_track_point_in_time
 from homeassistant.helpers.update_coordinator import (
     CoordinatorEntity,
     DataUpdateCoordinator,
 )
+import homeassistant.util.dt as dt_util
 
 from .const import DOMAIN, KEY_COORDINATOR, KEY_DEVICE_INFO, KEY_INVERTER
 
 # Sensor name of battery SoC
 BATTERY_SOC = "battery_soc"
 
+# Sensors that are reset to 0 at midnight.
```

Comment:
Please add some background in the comment to why this is needed and why it's safe to do.

Reply:
Done, I added a detailed explanation.